packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/MrwSerializationTypeProvider.cs

```diff
@@ -156,30 +188,37 @@ protected override MethodProvider[] BuildMethods()
         /// <returns>An array of <see cref="CSharpType"/> types that the model implements.</returns>
         protected override CSharpType[] BuildImplements()
         {
-            int interfaceCount = _iJsonModelObjectInterface != null ? 2 : 1;
-            CSharpType[] interfaces = new CSharpType[interfaceCount];
-            interfaces[0] = _iJsonModelTInterface;
-
+            List<CSharpType> interfaces = [_iJsonModelTInterface];
+            CSharpType? inheritedType = _model.Inherits;
+            if (inheritedType != null)
+            {
+                interfaces.Add(inheritedType);
+            }
             if (_iJsonModelObjectInterface != null)
             {
-                interfaces[1] = _iJsonModelObjectInterface;
+                interfaces.Add(_iJsonModelObjectInterface);
             }
 
-            return interfaces;
+            return interfaces.ToArray();
         }
 
         /// <summary>
         /// Builds the JSON serialization write method for the model.
         /// </summary>
         internal MethodProvider BuildJsonModelWriteMethod()
         {
+            var methodName = $"{nameof(IJsonModel<object>.Write)}Core";
```

Comment:
well I do not think this name has the possiblity of changing therefore maybe we could define this as a `const string`